aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -123,8 +140,10 @@ impl DeltaOp {
         // was +99 at some point. Now, if we merge some `d1` which is +2 with `d2`, we get
         // the result is +5. However, it should not have happened because `d2` should hit
         // +2+99 > 100 at some point in history and fail.
-        let shifted_max_positive = self.shifted_max_positive_by(&previous_delta)?;
-        let shifted_min_negative = self.shifted_min_negative_by(&previous_delta)?;
+        let shifted_max_achieved_positive =
```

Comment:
Cannot comment on the function declaration, how do overflows change as we merge deltas? e.g.:
d1 is +10 and overflows on +300 and d2 is +10 and overflows on +400, we result is +20 and overflows on +300

Reply:
This is actually important I think?

Reply:
As we aggregator delta from previous transactions, we don't change the overflow/underflow bounds of the current transaction right?

Reply:
I have the same question, it looks to me we should do a max/min of the four bounds? since squash is used for multiple sessions from the same txn iirc

Reply:
I was discussing with @gelash on this, it looks both used on squash and read, but I think the overflow bounds are important for both cases.
for reads we need to validate after shifting if it triggers overflow or not. for squash we need to take the max bounds I think

Reply:
The 4 values stored in the delta with (0, max_value) essentially determine the acceptable pre-image for which the behavior is intact.

Lots of things are missing here:
(1) we need to shift all 4 values of the new delta based on the previous value. this is because the "origin" for the current delta is now shifted.
(2) we then need to take min / max of the shifted bounds and the previous bounds, i.e. if the shifted overflew after adding 103 and the previous overflew after adding 101, then we store 101 in the min_added overflow and so on.
We then need to confirm that the new merged bounds are consistent, i.e. that max_added_no_overflow < max_added_overflow (similarly for the underflow side). 
The reason for this last part is to not assume that the speculative value used for "prediction" inside the session that generated the DeltaOp isn't the same (I believe if it was the same we could prove / assert that the bounds would end up consistent) - let's add a comment about this aspect too.
 
When we generate the new 4 values (if all is successful), We can also assert that the combined delta update value falls in between.

We can rewrite the shift functions as we need to, e.g. those can also do the min/max (but not assign until all the checks pass), right now it seems we do the max when we assign at the end of the function. And in general, feel free to refactor the utility functions, there should be improvement opportunities esp. since the flow and logic is evolving.